src/prefect/concurrency/sync.py

```diff
@@ -87,18 +93,31 @@ def main():
         _emit_concurrency_release_events(limits, occupy, emitted_events)
 
 
-def rate_limit(names: Union[str, List[str]], occupy: int = 1) -> None:
+def rate_limit(
+    names: Union[str, List[str]],
+    occupy: int = 1,
+    timeout_seconds: Optional[float] = None,
+    create_if_missing: Optional[bool] = True,
+) -> None:
     """Block execution until an `occupy` number of slots of the concurrency
     limits given in `names` are acquired. Requires that all given concurrency
     limits have a slot decay.
 
     Args:
         names: The names of the concurrency limits to acquire slots from.
         occupy: The number of slots to acquire and hold from each limit.
+        timeout_seconds: The number of seconds to wait for the slots to be acquired before
+            raising a `TimeoutError`. A timeout of `None` will wait indefinitely.
+        create_if_missing: If implicitly creating new limits, whether the limits are active.
```

Comment:
This docstring confuses me - isn't this flag for whether to implicitly create a new limit or not?

Reply:
Ah, good find - there is some baggage here that I didn't clean up properly from the completely different flag I added first and then replaced with `create_if_missing`. 😓 That flag controlled whether implicitly created limits were active. (They are currently always inactive with no way to control that at creation time. Yada yada.)